### PR TITLE
COMMERCE-4778 Fix Shipping Options search

### PR DESCRIPTION
**Pull request by alecsloan**

### CODE REVIEW

**Comment by alecsloan** [comment]
ci:test:sf

**Comment by alecsloan** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-shipping-engine-fixed-service/src/main/java/com/liferay/commerce/shipping/engine/fixed/internal/search/CommerceShippingFixedOptionIndexer.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+			SearchContext searchContext)
+		throws Exception {
+
+		super.postProcessSearchQuery(
+			searchQuery, fullQueryBooleanFilter, searchContext);
+
+		addSearchTerm(searchQuery, searchContext, Field.ENTRY_CLASS_PK, false);
+		addSearchTerm(
+			searchQuery, searchContext, "commerceShippingMethodId", false);
+		addSearchLocalizedTerm(searchQuery, searchContext, Field.NAME, true);
```
Comment (dsitu) [inline modules/apps/commerce/commerce-shipping-engine-fixed-service/src/main/java/com/liferay/commerce/shipping/engine/fixed/internal/search/CommerceShippingFixedOptionIndexer.java:92]: Sort constant chunks before string.

```suggestion
		addSearchLocalizedTerm(searchQuery, searchContext, Field.NAME, true);
		addSearchTerm(
			searchQuery, searchContext, "commerceShippingMethodId", false);
```

File: modules/apps/commerce/commerce-shipping-engine-fixed-service/src/main/java/com/liferay/commerce/shipping/engine/fixed/service/impl/CommerceShippingFixedOptionLocalServiceImpl.java [hunk 2]
```diff
@@ -154,4 +208,70 @@ public CommerceShippingFixedOption updateCommerceShippingFixedOption(
...
+			String keywords, int start, int end)
+		throws PortalException {
+
+		SearchContext searchContext = new SearchContext();
+
+		searchContext.setCompanyId(companyId);
+		searchContext.setGroupIds(new long[] {groupId});
+		searchContext.setKeywords(keywords);
+		searchContext.setStart(start);
+		searchContext.setEnd(end);
```
Comment (dsitu) [inline modules/apps/commerce/commerce-shipping-engine-fixed-service/src/main/java/com/liferay/commerce/shipping/engine/fixed/service/impl/CommerceShippingFixedOptionLocalServiceImpl.java:222]: sort
```suggestion
		searchContext.setCompanyId(companyId);
		searchContext.setEnd(end);
		searchContext.setGroupIds(new long[] {groupId});
		searchContext.setKeywords(keywords);
		searchContext.setStart(start);
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by alecsloan** [comment]
@dsitu updated thanks!

**Comment by alecsloan** [comment]
ci:test:sf

**Review by dsitu — APPROVED** [review]

**Comment by marco-leo** [comment]
ci:forward

**Pull request closed without merging**